### PR TITLE
Split of "#111 Fix lane count calculation"

### PR DESCRIPTION
**Pull request by droogmic**
This is what https://github.com/a-b-street/osm2lanes/pull/111 would look like applied to:
- https://github.com/a-b-street/osm2lanes/pull/141
- https://github.com/a-b-street/osm2lanes/pull/145

This is getting to a reviewable size.

### CODE REVIEW

File: rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs [hunk 1]
```diff
@@ -255,6 +296,7 @@ fn lanes_bus(
     Ok(())
 }
 
+//noinspection ALL
```
Comment (droogmic) [inline rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs:299]: hmm

**Comment by droogmic** [comment]
I think this is getting to a reasonable size, if you approve this I will chain merge https://github.com/a-b-street/osm2lanes/pull/141, https://github.com/a-b-street/osm2lanes/pull/145 and finally this one.

File: rust/osm2lanes/src/transform/tags_to_lanes/mod.rs [hunk 2]
```diff
@@ -244,11 +209,24 @@ impl RoadBuilder {
...
+        let bus_lane_counts = BusLanesCount::from_tags(tags, locale, oneway, warnings)?;
+        let centre_turn_lanes = CentreTurnLaneScheme::new(tags, oneway, locale, warnings);
+        let lanes = Counts::new(
+            tags,
+            oneway,
+            &centre_turn_lanes,
+            &bus_lane_counts,
+            locale,
+            warnings,
+        );
```
Comment (BudgieInWA) [inline rust/osm2lanes/src/transform/tags_to_lanes/mod.rs:228]: I love the idea that the road builder is responsible for all of this. It's pretty clearly the right spot.

Combined with "builder" in the name, it makes me think of something like this. The call graph becomes the things that describes the dependencies, which is interesting.

```rs
struct RoadBuilder {
    // ...
    pub oneway: Oneway,
    pub bus_lane_counts: BusLanesCount,
    pub centre_turn_lanes: CentreTurnLaneScheme,
    pub lane_counts: Counts,

    pub locale: Locale,
    warnings: RoadWarnings,
}

impl  RoadBuilder {
	pub fn from_tags(tags: &Tags, locale: &Locale) -> Result {
	      let road = Self::local_default(locale);
	      road.oneway = Oneway::from(tags.is_any("oneway", &["yes", "-1"]) || tags.is("junction", "roundabout"));
	      road.bus_lane_counts = BusLanesCount::from_tags(tags, &self)?;
	      road.centre_turn_lanes = CentreTurnLaneScheme::from_tags(tags, &self)?;
	      road.lane_counts = Counts::from_tags(tags, &self);
	}
	pub fn add_warning(...){...}
}

// ::default() for everything...
```

File: rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs (hunk 1 above)
Comment (BudgieInWA) [reply rust/osm2lanes/src/transform/tags_to_lanes/modes/bus.rs]: Whoops, this was a side-effect of trying to fix that reported error. Well spotted.

**Pull request merged**